https://bugs.gentoo.org/686012
In file included from ../inc/cc++/thread.h:50,
                 from thread.cpp:41:
../inc/cc++/string.h:734:35: error: friend declaration of ‘std::istream& getline(std::istream&, ost::String&, char, size_t)’ specifies default arguments and isn’t a definition [-fpermissive]
  734 |     friend __EXPORT std::istream &getline(std::istream &is, String &str, char delim = '\n', size_t size = 0);
      |                                   ^~~~~~~

--- a/inc/cc++/string.h
+++ b/inc/cc++/string.h
@@ -58,6 +58,9 @@
 
 class MemPager;
 
+class String;
+__EXPORT std::istream &getline(std::istream &is, String &str, char delim = '\n', size_t size = 0);
+
 /**
  * This is a generic and portable string class.  It uses optimized
  * memory allocation strategies to efficiently handle smaller string
@@ -731,7 +734,7 @@
      * @param delim deliminator to use.
      * @param size optional size limitor.
      */
-    friend __EXPORT std::istream &getline(std::istream &is, String &str, char delim = '\n', size_t size = 0);
+    friend std::istream &getline(std::istream &is, String &str, char delim, size_t size);
 
     /**
      * Stream the content of our string variable directly to a C++
